### PR TITLE
Add livestreamer rank

### PR DESCRIPTION
**Pull request by dentmaged**
@TiTi-418 suggested this so I implemented it.

Livestreamers may not chat in the global chat until the game has ended, nor join a team. However, if you give them the permission ```cardinal.livestreamer.bypass```, they can join a team and speak in global chat.

### CODE REVIEW

File: src/main/java/in/twizmwaz/cardinal/module/modules/permissions/PermissionModule.java [hunk 1]
```diff
@@ -172,8 +193,9 @@ public void onPlayerNameUpdate(PlayerNameUpdateEvent event) {
             stars += ChatColor.GOLD + star;
         } else if (isMod(event.getPlayer().getUniqueId())) {
             stars += ChatColor.RED + star;
-        }
-        if (devs.contains(event.getPlayer().getUniqueId())) {
```
Comment (iEli2tyree011) [inline src/main/java/in/twizmwaz/cardinal/module/modules/permissions/PermissionModule.java]: Why did you remove this `if` statement? That makes it so someone can't have both an op star and a dev star (or any other two stars).

**Comment by SungMatt** [comment]
Why do we need a livestreamer rank?

File: src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java [hunk 2]
```diff
@@ -52,12 +55,20 @@ public Inventory getTeamPicker(Player player) {
                 totalPlayers += team.size();
             }
         }
```
Comment (iEli2tyree011) [inline src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java]: When you're doing the below changes, they only show or go away when the player's inventory is reset. If I un-livestreamer someone, they still keep the "livestreamer inventory".

File: src/main/java/in/twizmwaz/cardinal/module/modules/permissions/PermissionModule.java (hunk 1 above)
Comment (dentmaged) [inline src/main/java/in/twizmwaz/cardinal/module/modules/permissions/PermissionModule.java]: What? I thought I added else if!

File: src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java (hunk 2 above)
Comment (dentmaged) [inline src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java]: Okay, I'll update their inventory!

Comment (iEli2tyree011) [inline src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java]: You shouldn't update their inventory, because they aren't switching teams. Maybe try getting the affected items and changing their lore and display name to the right ones. As @SungMatt said though, there isn't much reason to have a livestreamer rank.

**Comment by dentmaged** [comment]
@SungMatt @iEli2tyree011 Non-profitable public tournaments?

File: src/main/java/in/twizmwaz/cardinal/command/JoinCommand.java [hunk 3]
```diff
@@ -20,6 +24,12 @@
     @Command(aliases = {"join", "play"}, desc = "Join a team.", usage = "[team]")
     public static void join(final CommandContext cmd, CommandSender sender) throws CommandException {
         if (sender instanceof Player) {
+            Player player = (Player) sender;
```
Comment (iEli2tyree011) [inline src/main/java/in/twizmwaz/cardinal/command/JoinCommand.java]: If someone is made a livestreamer while playing a match, aren't they locked into their team?

Comment (dentmaged) [inline src/main/java/in/twizmwaz/cardinal/command/JoinCommand.java]: @iEli2tyree011 Yep, fixed!

File: src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java (hunk 2 above)
Comment (dentmaged) [inline src/main/java/in/twizmwaz/cardinal/module/modules/teamPicker/TeamPicker.java]: @iEli2tyree011 It shouldn't do that. As when you click with the hat, it creates a new inventory and shows it the player.

**Comment by twizmwazin** [comment]
Closing as this really isn't needed. In the future, expect some form of custom ranks with custom per-rank permissions.

**Comment by dentmaged** [comment]
@twizmwazin Can I make that now? Or when Cardinal v1 is released?

**Comment by twizmwazin** [comment]
@dentmaged It is very likely that there will ever be an official '1.0' release. I intend on keeping a rolling release, as I want users to have the latest features and bug fixes. You can take a stab at custom ranks if you want, although I make **no** promises that I would merge it.

**Comment by dentmaged** [comment]
@twizmwazin Okay! I'll begin work after I have done my revision.

**Comment by TiTi-418** [comment]
@dentmaged Wow, thanks :)
